Rx/v2/src/rxcpp/rx-observable.hpp

```diff
@@ -263,27 +263,107 @@ class blocking_observable
         return blocking_subscribe(source, std::forward<ArgN>(an)...);
     }
 
+    /*! Return the first item emitted by this blocking_observable, or throw an std::runtime_error exception if it emits no items.
+
+        \return  The first item emitted by this blocking_observable.
+
+        \note  If the source observable calls on_error, the raised exception is rethrown by this method.
+
+        \sample
+        When the source observable emits at least one item:
+        \snippet blocking_observable.cpp blocking first sample
+        \snippet output.txt blocking first sample
+
+        When the source observable is empty:
+        \snippet blocking_observable.cpp blocking first empty sample
+        \snippet output.txt blocking first empty sample
+    */
     T first() {
-        rxu::maybe<T> result;
```

Comment:
I think this change to first() should be reverted. the old code is a lot more efficient than using the non-blocking operator. just adding
```cpp
        if (result.empty())
            throw rxcpp::empty_error("No elements");
```
as was done in last() will cover it since blocking_subscribe will rethrow the error.